aspnetcore/fundamentals/use-http-context/samples/Program.cs

```diff
@@ -0,0 +1,174 @@
+{
+    var filePath = Path.Combine(config["StoredFilesPath"], Path.GetRandomFileName());
+
+    await using var writeStream = File.Create(filePath);
+    await context.Request.Body.CopyToAsync(writeStream);
+});
+
+app.Run();
+#endregion
+#elif RequestBuffering
+#region snippet_RequestBuffering
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+app.Use(async (context, next) =>
+{
+    context.Request.EnableBuffering();
+    await ReadRequestBody(context.Request.Body);
+    context.Request.Body.Position = 0;
+    
+    await next.Invoke();
+});
+
+app.Run();
+#endregion
+#elif ResponseHeaders
+#region snippet_ResponseHeaders
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+app.MapGet("/", (HttpResponse response) =>
+{
+    response.Headers.CacheControl = "no-cache";
+    response.Headers["x-custom-header"] = "Custom value";
+
+    return Results.File(File.OpenRead("helloworld.txt"));
+});
+
+app.Run();
+#endregion
+#elif ResponseTrailers
+#region snippet_ResponseTrailers
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+app.MapGet("/", (HttpResponse response) =>
+{
+    // Write body
+    response.WriteAsync("Hello world");
+    response.AppendTrailer("trailername", "TrailerValue");
+});
+
+app.Run();
+#endregion
+#elif ResposeBody
+#region snippet_ResponseBody
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+app.MapPost("/downloadfile", async (IConfiguration config, HttpContext context) =>
+{
+    var filePath = Path.Combine(config["StoredFilesPath"], "helloworld.txt");
+
+    await using var fileStream = File.OpenRead(filePath);
+    await fileStream.CopyToAsync(context.Response.Body);
+});
+
+app.Run();
+#endregion
+#elif RequestAborted
+#region snippet_RequestAborted
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+var httpClient = new HttpClient();
+app.MapPost("/books/{bookId}", async (int bookId, HttpContext context) =>
+{
+    var stream = await httpClient.GetStreamAsync(
+        $"http://consoto/books/{bookId}.json", context.RequestAborted);
+
+    // Proxy the response as JSON
+    return Results.Stream(stream, "application/json");
+});
+
+app.Run();
+#endregion
+#elif Abort
+#region snippet_Abort
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+var httpClient = new HttpClient();
+app.MapPost("/", async (HttpContext context) =>
+{
+    var task = context.Request.ReadFromJsonAsync<CustomType>(); 
+    
+    var cts = new CancellationTokenSource();
+    var timeoutTask = Task.Delay(TimeSpan.FromSeconds(10), cts.Token);
+    
+    var completedTask = await Task.WhenAny(task, timeoutTask);
+    if (completedTask == task)
+    {
+        cts.Cancel();
+
+        var requestMessage = await task;
+        // Use request message.
+    }
+    else
+    {
+        // Abort the request.
+        context.Abort();
+    }
```

Comment:
I understand you were trying to show how to use Abort, but that's not code we'd want anyone to copy.
```suggestion
    using var cts = new CancellationTokenSource(TimeSpan.FromSeconds(10));
    var result = await context.Request.ReadFromJsonAsync<CustomType>(cts.Token);
```

How about:
```
if (RequestAppearsMalicious(context))
{
   context.Abort(); // Malicious requests don't even deserve an error response (e.g. 400).
   return;
}
```

Reply:
Ok, that makes sense. I was struggling to come up with a situation that someone would want to call it.